### ISSUE TITLE
Duplicate Autosuggestions

### ISSUE THREAD

**Issue by jackcarlisle** [numo-labs/isearch-ui#268]
When searching for a tag there are still some duplicate values coming back and being displayed beneath the search bar:
<img width="929" alt="screen shot 2016-06-13 at 11 58 56" src="https://cloud.githubusercontent.com/assets/12450298/16005263/36ec7d06-315e-11e6-8f22-34360c489707.png">

**Comment by tomgco** [comment]
These are duplicated due to incorrect data in the tagging system.

They are actually two distinct locations linked to different areas.

We could link the two together and re-run @jruts' de-dupe script?

**Comment by nelsonic** [comment]
Issue still persists in "Prod" today:
<img width="677" alt="isearch-prod-duplicate-autosuggestions" src="https://cloud.githubusercontent.com/assets/194400/16541436/d41e0650-407a-11e6-8d67-683b9d24cd47.png">

**Comment by lennym** [comment]
I *think* I can see the root cause of this. taggable searcher does a query to match the tag id and the display name, so if the string hits both then it comes back twice.

Will investigate...

**Comment by lennym** [comment]
There's more than one issue in play here... 

`Spanisches Dorf "Pueblo Espanol"` is a duplicate tag (`geo:geonames.6946451` and `geo:geonames.6947342`) so we will want to decide if that is intentional and if so, how to resolve it.

`Spa & Relax` is `marketing:term.spa` being erroneously returned twice though. Will look further into why that is.

**Comment by lennym** [comment]
Having reviewed the `Spa & Relax` tag, the reason for duplication is the aliasing of search terms. The tag `marketing:term.spa` is set to show for searches for:

* Spa & Relax
* Spa Together
* Spa
* ... and others

This means that entering "spa" will match all of these, since they are indexed into elasticsearch as unique entries, with different names, but resolving to the same tagid. Hence the tag appears multiple times in sutosuggest.

https://github.com/numo-labs/taggable-searcher/pull/47 resolves this by ensuring that any tag id is only represented at most once in any autosuggest result.

Note: the duplicate tag for `Spanisches Dorf "Pueblo Espanol"` is a separate issue, and is not resolved.

**Comment by tomgco** [comment]
> numo-labs/taggable-searcher#47 resolves this by ensuring that any tag id is only represented at most once in any autosuggest result.

Because this is an npm dependency we will need to tag and deploy a new version to `npm`.

Once this is done we will then need to update the lambda's which use taggable-searcher in the auto-completer.

**Comment by lennym** [comment]
@tomgco I think it needs you or @nelsonic to publish a new version.

If you would be so kind...

**Comment by tomgco** [comment]
@lennym I noticed that you now have credentials to do this yourself?

**Comment by nelsonic** [comment]
Is this still an issue?

**Comment by emiledskar** [comment]
We still have this issue for the New York tag. Both in prod and ci I get two suggested New York tags. The tags has different content bound to them as well. 
<img width="427" alt="screen shot 2016-07-20 at 14 37 00" src="https://cloud.githubusercontent.com/assets/8721221/16986560/7623f800-4e87-11e6-98ed-c47bb342260d.png">

**Comment by emiledskar** [comment]
Found that "Dubai" also has duplicated tags. 
![image](https://cloud.githubusercontent.com/assets/8721221/17174142/9c207fb0-53ff-11e6-85b8-38fe22792628.png)

**Comment by Kumjami** [comment]
@lennym how is it going? are you working on it?

**Comment by lennym** [comment]
@Kumjami I'm not working on anything related to this, no. As far as I can see it's a content rather than a code issue.

**Comment by Kumjami** [comment]
@lennym who should take care about content? I mean, should we change content under our criteria? or is someone responsible for that? (Roy, @emiledskar, Rasmus, Martin...)

**Comment by tomgco** [comment]
Looking into Dubai this isn't a content issue.

The Geo for one of the Dubai's is correct: geo:geonames.292223

However we seem to have a geo for a non-existent geo: geo:geonames.292224 

And this will be why we are seeing a duplicate for Dubai.

**Comment by emiledskar** [comment]
For the New York tag it seems that some articles are using a removed New York tag (geo:geonames.5128638) because I can't find that tag in Taggable UI.

**Comment by lennym** [comment]
I can see geo:geonames.292224 in S3 - I don't know why it's not appearing in taggable-ui.

It may need to be re-deployed with the latest `taggable-searcher` version.

**Comment by lennym** [comment]
This looks to be a problem with the elasticsearch indexer not creating an entry for `context: 'taggy'` which the taggable-ui specifies. This seems to apply to both `geo:geonames.5128638` (New York) and `geo:geonames.292224` (Dubai)

There are elasticsearch records for all other contexts applicable, but not that one. I have no idea why.

Either way, it looks like a good time for my taggable re-indexer script to come out again.

**Comment by tomgco** [comment]
@lennym Will this take care of the issue at hand and dedupe Dubai and New York, or will this just correctly show the missing geo's in taggable-ui?

**Comment by lennym** [comment]
The latter.

There are 106 geo tags which aren't correctly indexed and won't show up in taggable-ui. I'm about to force those to be re-indexed. 

I'll also push the re-indexing scripts to github as they only exist on my computer right now.

**Comment by emiledskar** [comment]
As of now I will correct these duplicated tags by simply deactivating all but one of them.

**Comment by lennym** [comment]
@tomgco @emiledskar I've published the code for the indexer I've been using to do batch re-indexing/querying on `taggable`. It's at https://github.com/numo-labs/taggable-indexer if you want to take a look.

The examples directory contains some of the recent queries I've run for @emiledskar.

**Comment by emiledskar** [comment]
@lennym Is there any way that I could find out all tags that has a search term with X label? Sometimes it is nearly impossible to find all the duplicated tags by searching through the Taggable UI.

**Comment by lennym** [comment]
@emiledskar Yes - I ran this code earlier when looking for things with a label of `search` - https://github.com/numo-labs/taggable-indexer/blob/master/examples/check-search.js#L26

You can either run that yourself or if you let me know what label you're after then I can do it for you.
